repository/src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/NodePermissionAssessor.java

```diff
@@ -145,14 +149,16 @@ public boolean shouldQuitChecks()
     {
         boolean result = false;
         
-        if (checksPerformed >= maxPermissionChecks)
+        if (checksPerformed >= Double.valueOf(maxPermissionChecks) + 1)
         {
             result = true;
+            logger.warn("Maximum permission checks exceeded (" + maxPermissionChecks + ")");
         }
         
         if ((System.currentTimeMillis() - startTime) >= maxPermissionCheckTimeMillis)
         {
             result = true;
+            logger.warn("Maximum permission checks time exceeded (" + maxPermissionCheckTimeMillis + ")");
         }
         
         return result;
```

Comment:
Would push for early exit here removing the "result" variable

Reply:
Done.